lib/graphql/subscriptions/anycable_subscriptions.rb

```diff
@@ -86,11 +120,18 @@ def deliver(subscription_id, result)
       # Save query to "storage" (in redis)
       def write_subscription(query, events)
         context = query.context.to_h
-        subscription_id = context[:subscription_id] ||= build_id
+        subscription_id = context.delete(:subscription_id) || build_id
         channel = context.delete(:channel)
-        stream = context[:action_cable_stream] ||= SUBSCRIPTION_PREFIX + subscription_id
+        stream = context.delete(:action_cable_stream) || SUBSCRIPTION_PREFIX + subscription_id
         channel.stream_from(stream)
 
+        # Maybe we can eliminate intermediate step in (topic→fingerprint→subscription_id) and stream right by event fingerprint?
```

Comment:
Yeah, for broadcastable subscriptions we can do that; `stream_from(event.fingerprint)` and `broadcast(event.fingerprint)` should work (and we probably would like to add a prefix to stream names).

In this case we don't need to store anything in Redis, I guess.

Reply:
We still need to store mapping from `event.topic` to `event.fingerprint` because at the time when `Schema.subscriptions.trigger` is called somewhere in application we have only it.

And we need to store graphql query with metadata somewhere (under `event.fingerprint`) to reevaluate it and broadcast results.

Reply:
Theoretically we can avoid storing anything in Redis by using [state accessors](https://docs.anycable.io/#/v1/rails/channels_state?id=using-state_attr_accessor) but I can't imagine for now how it can be implemented. Kinda broadcast internal message with event topic to Websocket servers and let them go to RPC server to reevalute query once and broadcast results to each server clients maybe